File: cpp/0226翻转二叉树.cpp
#include"mybitree.h"
#include<deque>

// 递归
TreeNode *invertTree(TreeNode *root)
{
    if(!root||(!root->left&&!root->right))
        return root;

    TreeNode *lc = invertTree(root->left);
    TreeNode *rc = invertTree(root->right);

    root->left = rc;
    root->right = lc;

    return root;
}

// 迭代
TreeNode* invertTree(TreeNode*root)
{
    deque<TreeNode *> nodes;
    nodes.push_back(root);

    while(!nodes.empty())
    {
        TreeNode *crt = nodes.front();
        nodes.pop_front();
        if(crt!=nullptr)
        {
            TreeNode *temp = crt->left;
            crt->left = crt->right;
            crt->right = temp;
            nodes.push_back(crt->left);
            nodes.push_back(crt->right);
        }

    }

    return root;
}

File: python/main/0226翻转二叉树.py
from binarytree import TreeNode

# 递归
def invertTree(root: TreeNode) -> TreeNode:
    
    if(not (root and any([root.left,root.right]))):
        return root
    
    lc=invertTree(root.left)
    rc=invertTree(root.right)

    root.left=rc
    root.right=lc

    return root

from collections import deque

# 迭代
def invertTree1(root: TreeNode) -> TreeNode:
    nodes=deque([root])

    while(nodes):
        crt=nodes.pop()

        if(crt):
            nodes.append(crt.left)
            nodes.append(crt.right)

            crt.left,crt.right=crt.right,crt.left
    return root